moxy-compiler/src/main/java/moxy/compiler/Util.java

```diff
@@ -224,9 +225,8 @@ public static String decapitalizeString(String string) {
     public static <T> boolean equalsBy(
         Collection<T> first,
         Collection<T> second,
-        BiFunction<T, T, Boolean> predicate) {
+        @NotNull BiFunction<T, T, Boolean> predicate) {
```

Comment:
You've removed an explicit runtime check, and if predicate is null, this will crash with a non-obvious NPE. Maybe just write this precondition manually?

Reply:
It is used internally by library in single place, annotation will warn about this parameter being not null. And it had no precondition check for long time, it was added in one of the recent PRs. So i dont think there should be any check